std/string.d

```diff
@@ -2887,14 +2890,69 @@ if (isConvertibleToString!Range)
     assert(testAliasedString!stripLeft("  hello"));
 }
 
+/// Ditto
+auto stripLeft(Range, Char)(Range input, const(Char)[] chars)
+if (isForwardRange!Range && isSomeChar!(ElementEncodingType!Range) &&
+    !isConvertibleToString!Range && isSomeChar!Char)
+{
+    for (; !input.empty; input.popFront)
+    {
+        if (chars.indexOf(input.front) == -1)
+            break;
+    }
+    return input;
+}
+
+///
+@safe pure unittest
+{
+     assert(stripLeft("     hello world     ", " ") ==
+           "hello world     ");
+    assert(stripLeft("xxxxxhello world     ", "x") ==
+           "hello world     ");
+    assert(stripLeft("xxxyy    hello world     ", "xy ") ==
+           "hello world     ");
+
+    import std.array : array;
+    import std.utf : byChar, byWchar, byDchar;
+
+    assert(stripLeft("  xxxyy hello world     "w.byChar, "xy ").array ==
+           "hello world     ");
+
+    assert(stripLeft("\u2028\u2020hello world\u2028"w.byWchar,
+                     "\u2028").array == "\u2020hello world\u2028");
+    assert(stripLeft("\U00010001hello world"w.byWchar, " ").array ==
+           "\U00010001hello world"w);
+    assert(stripLeft("\U00010001 xyhello world"d.byDchar,
+                     "\U00010001 xy").array == "hello world"d);
+
+    assert(stripLeft("\u2020hello"w, "\u2020"w) == "hello"w);
+    assert(stripLeft("\U00010001hello"d, "\U00010001"d) == "hello"d);
+    assert(stripLeft(" hello ", "") == " hello ");
+}
+
+auto stripLeft(Range, Char)(auto ref Range str, const(Char)[] chars)
```

Comment:
Public and undocumented not good. Please merge constraints and bodies with the overload above.

Reply:
I followed similar to existing `stripLeft(Range)(auto ref Range str)`. Confused because none of the functions with `auto ref` input are documented in this file. Please clarify.

Reply:
We pursue lifting the quality of code overall, and in the process of doing that it's possible that new code is subject to a higher bar than existing code. Two standards arose from experience:

* All public symbols must be documented
* Prefer fewer overloads

I understand this can be frustrating - please bear with us. Thanks!

Reply:
No problem, happy to work on the changes, just needed a clarification. Thanks